src/secondary_threads/parse_packets.rs

```diff
@@ -45,152 +50,206 @@ pub fn parse_packets(
+                                // waiting for a previously requested rDNS resolution
+                                // update the corresponding waiting address data
+                                info_traffic
+                                    .addresses_waiting_resolution
+                                    .entry(address_to_lookup)
+                                    .and_modify(|data_info| {
+                                        data_info.add_packet(
                                             exchanged_bytes,
                                             new_info.traffic_direction,
                                         );
                                     });
-                                }
                             }
-
-                            //increment the packet count for the sniffed app protocol
-                            info_traffic
-                                .app_protocols
-                                .entry(application_protocol)
-                                .and_modify(|data_info| {
-                                    data_info
+                            (_, true) => {
+                                // rDNS already resolved
+                                // update the corresponding host's data info
+                                let host = info_traffic
+                                    .addresses_resolved
+                                    .get(&address_to_lookup)
+                                    .unwrap()
+                                    .1
+                                    .clone();
+                                info_traffic.hosts.entry(host).and_modify(|data_info_host| {
+                                    data_info_host
+                                        .data_info
                                         .add_packet(exchanged_bytes, new_info.traffic_direction);
-                                })
-                                .or_insert(DataInfo::new_with_first_packet(
-                                    exchanged_bytes,
-                                    new_info.traffic_direction,
-                                ));
+                                });
+                            }
                         }
+
+                        //increment the packet count for the sniffed app protocol
+                        info_traffic
+                            .app_protocols
+                            .entry(application_protocol)
+                            .and_modify(|data_info| {
+                                data_info.add_packet(exchanged_bytes, new_info.traffic_direction);
+                            })
+                            .or_insert(DataInfo::new_with_first_packet(
+                                exchanged_bytes,
+                                new_info.traffic_direction,
+                            ));
                     }
                 }
             }
         }
     }
 }
+
```

Comment:
Adding a helper function with slice length check and AF number checking (for other comments bellow):

```suggestion
fn try_from_null_frame<'a>(frame: &'a [u8]) -> Result<PacketHeader<'a>, ReadError> {
    if frame.len() < 4 {
        return Err(ReadError::UnexpectedEndOfSlice(packet.len()));
    }
    let is_ip_null_frame_af_num = {
        // based on https://wiki.wireshark.org/NullLoopback.md (2023-12-31)
        fn matches(value: u32) -> bool {
            match value {
                // 2 = IPv4 on all platforms
                // 24, 28, or 30 = IPv6 depending on platform
                2 | 24 | 28 | 30 => true,
                _ => false,
            }
        }
        let s = &frame[..4];
        let b = [s[0], s[1], s[2], s[3]];
        // check both big endian and little endian representations
        // as some OS'es use native endianess and others use big endian
        matches(u32::from_le_bytes(b)) || matches(u32::from_be_bytes(b))
    }
    
    if is_ip_null_frame_af_num {
        PacketHeaders::from_ip_slice(&packet[4..])
    } else {
        Ok(PacketHeaders{
            payload: &[],
            ..Default::default()
        })
    }
}
```

Reply:
Why not returning an `Err` in case `is_ip_null_frame_af_num` is false?

Reply:
I'm not checking errors (just discarding them) so I think it's reasonable to put there one of the `ReadError`s even if not meaningful for this specific case.

Reply:
I think that
``` rust
         Err(ReadError::IoError(std::io::Error::new(
            ErrorKind::InvalidData,
            "Invalid AF_INET / AF_INET6 value",
        )))
```
is the most appropropriate one

Reply:
That is also fine. I just had no error in mind to use and just assumed that it would be filtered out by the upper layer anyways.